### PR TITLE
Browser Language Preference Needs To Be Honored

### LINKED ISSUE #15744: Browser Language Preference Needs To Be Honored
##### **Overview of the issue**

When Spanish is on the top of the language setting in my browser, my app still shows the content in the default language, English although Spanish is a language in my app.

##### **Motivation for or Use Case**

The browser language setting needs to be honored.

##### **Reproduce the error**

Create an app with multiple languages such as English as the default language and Spanish. Set Spanish on the top of your browser language setting. Access the app and it will show in English.

##### **JHipster configuration**
```
{
  "generator-jhipster": {
    "applicationType": "monolith",
    "baseName": "myapp",
    "jhipsterVersion": "7.1.0",
    "skipClient": false,
    "skipServer": false,
    "skipUserManagement": false,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "testFrameworks": ["cypress", "gatling"],
    "blueprints": [],
    "otherModules": [],
    "pages": [],
    "creationTimestamp": 1625095262715,
    "serviceDiscoveryType": false,
    "reactive": false,
    "authenticationType": "jwt",
    "packageName": "com.abc.myapp",
    "serverPort": "8080",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "buildTool": "gradle",
    "serverSideOptions": [],
    "websocket": false,
    "searchEngine": false,
    "messageBroker": false,
    "enableSwaggerCodegen": false,
    "clientFramework": "vue",
    "withAdminUi": true,
    "clientTheme": "cosmo",
    "clientThemeVariant": "primary",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "packageFolder": "com/abc/myapp",
    "jwtSecretKey": "NThkZDg5YjE3OTkxNGU2YjUzNjAwODMwZjY4ODRiYmIyOWVkN2Q0NmZhMjZhNDU4MmQyYTQ5NDcxOWZlODRjY2ZhYWE5NjNjZWJkZDUyYzljYTU0NmU0OGExZDY4N2EwZjljZWQwMTliYWU3NGZiZTQ3NDQ1MGIwOTNjZDU3NmE=",
    "devServerPort": 9060,
    "clientPackageManager": "npm",
    "languages": ["en", "es"]
  }
}
```

##### **Browsers and Operating System**

- [ ] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by vw98075**
Fix #15744

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/client/templates/vue/src/main/webapp/app/core/jhi-navbar/jhi-navbar.component.ts.ejs [hunk 1]
```diff
@@ -105,4 +107,18 @@ export default class JhiNavbar extends Vue {
   public get inProduction() : boolean {
     return (this.$store.getters.activeProfiles.indexOf('prod') > -1);
   }
+
+<%_ if (enableTranslation) { _%>
+  findLanguage(): string {
+    const languageKeys = Object.keys(JSON.parse(JSON.stringify(this.languages)));
```
Comment (deepu105) [inline generators/client/templates/vue/src/main/webapp/app/core/jhi-navbar/jhi-navbar.component.ts.ejs:113]: why is JSON parse/stringify here, languages is already a proper object

Comment (deepu105) [reply generators/client/templates/vue/src/main/webapp/app/core/jhi-navbar/jhi-navbar.component.ts.ejs:113]: I dont think this is right, with this logic you could end up with wrong versions for many languages, I'll rather do just the below

```
const currentLanguage = Object.keys(this.languages).includes(navigator.language) ? navigator.language : this.currentLanguage;
this.translationService().refreshTranslation(currentLanguage);
```

**Review by deepu105 — CHANGES_REQUESTED** [review]

File: generators/client/templates/vue/src/main/webapp/app/core/jhi-navbar/jhi-navbar.component.ts.ejs (hunk 1 above)
Comment (vw98075) [reply generators/client/templates/vue/src/main/webapp/app/core/jhi-navbar/jhi-navbar.component.ts.ejs]: Thanks for your feedback. I test your code and the outcome is the same for our test cases. I have changed the code in the branch last Friday. I don't know whether that is the right way to update a pull request or not.

**Review by deepu105 — APPROVED** [review]
Looks good

**Comment by Tcharl** [comment]
Can you sign the CLA so that we can go further? If you do welcome!

**Comment by mraible** [comment]
Is this one ready?

**Comment by DanielFran** [comment]
@mraible It is, but only for Vue

**Comment by Tcharl** [comment]
There's already many feature gap between frontend: if you agree let's not block single frontend tech feature

**Comment by mraible** [comment]
@vw98075 Is this ready to be merged? If so, please mark it ready for review.

**Comment by vw98075** [comment]
@mraible Hi, Matt, I believe that @deepu105 already reviewed it. Not sure about any missing official step though.

**Comment by mraible** [comment]
@vw98075 It looks like the CLA hasn't been signed yet. That's why CI isn't succeeding.

**Pull request merged**